### ISSUE TITLE
Accessibility for screen readers

### ISSUE THREAD

**Issue by darmro** [vaadin-component-factory/vcf-toggle-button#3]
This component is (IMO) commonly used. We're using it (Vaadin 22) and it's great. After all, toggle is one of the important elements of the UI interface.
But it turns out (notifications from our blind users) that it is completely invisible to screen readers (eg. NVDA), and it is impossible to tick its value by the blind. There is of course a way to select this field via keyboard but screen readers say nothing about this field because the internal checkbox field takes this focus and there is no aria-label, no title on it. So the user doesn't know when he is about to use the keyboard to tick the value. 
Setting aria-label on the root tag is not enough because the root tag takes no focus. We can set tabindex to it, but then user thinks he can set the value via keyboard, but no, because focus is not on the checkbox. 
We managed to somehow improve it, i.e. through javascript set the keyevent to the space key when focus is on the root tag and set the checkbox value on / off. And it works. But it is a hack way in javascript to take shadow element (this.shadowRoot.querySelector('vaadin-checkbox')).

I think it would be very nice to add this component to the list of changes regarding accessibility, which were started with Vaadin 22 :)
[https://vaadin.com/blog/upcoming-accessibility-improvements](https://vaadin.com/blog/upcoming-accessibility-improvements)
